pkg/util/cloudfoundry/garden_linux.go

```diff
@@ -0,0 +1,130 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-Present Datadog, Inc.
+
+package cloudfoundry
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"code.cloudfoundry.org/garden"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/containers/providers"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	// Needs ContainerImpl, so make sure it is init
+	_ "github.com/DataDog/datadog-agent/pkg/util/containers/providers/cgroup"
```

Comment:
The idea is to put this import where we know the OS we'll be running on, usually around the `main` file, if possible